model/LGCN-TensorFlow/LightGCN.py

```diff
@@ -78,20 +80,32 @@ def __init__(self, data_config, pretrain_data):
         
         
         with tf.name_scope('TRAIN_ACC'):
-            self.train_rec_first = tf.placeholder(tf.float32)
-            #record for top(Ks[0])
-            tf.summary.scalar('train_rec_first', self.train_rec_first)
-            self.train_rec_last = tf.placeholder(tf.float32)
-            #record for top(Ks[-1])
-            tf.summary.scalar('train_rec_last', self.train_rec_last)
-            self.train_ndcg_first = tf.placeholder(tf.float32)
-            tf.summary.scalar('train_ndcg_first', self.train_ndcg_first)
-            self.train_ndcg_last = tf.placeholder(tf.float32)
-            tf.summary.scalar('train_ndcg_last', self.train_ndcg_last)
-            self.train_precision_first = tf.placeholder(tf.float32)
-            tf.summary.scalar('train_precision_first', self.train_precision_first)
-            self.train_precision_last = tf.placeholder(tf.float32)
-            tf.summary.scalar('train_precision_last', self.train_precision_last)
+            if self.eval_type == 'foldout':
+                self.train_rec_first = tf.placeholder(tf.float32)
+                #record for top(Ks[0])
+                tf.summary.scalar('train_rec_first', self.train_rec_first)
+                self.train_rec_last = tf.placeholder(tf.float32)
+                #record for top(Ks[-1])
+                tf.summary.scalar('train_rec_last', self.train_rec_last)
+                self.train_ndcg_first = tf.placeholder(tf.float32)
+                tf.summary.scalar('train_ndcg_first', self.train_ndcg_first)
+                self.train_ndcg_last = tf.placeholder(tf.float32)
+                tf.summary.scalar('train_ndcg_last', self.train_ndcg_last)
+                self.train_precision_first = tf.placeholder(tf.float32)
+                tf.summary.scalar('train_precision_first', self.train_precision_first)
+                self.train_precision_last = tf.placeholder(tf.float32)
+                tf.summary.scalar('train_precision_last', self.train_precision_last)
+                self.train_precision_last = tf.placeholder(tf.float32)
+            
+            elif self.eval_type == 'loo':
+                self.train_hitrate_first = tf.placeholder(tf.float32)
+                tf.summary.scalar('train_hitrate_first', self.train_hitrate_first)
+                self.train_mrr_first = tf.placeholder(tf.float32)
+                tf.summary.scalar('train_mrr_first', self.train_mrr_first)
+                self.train_hitrate_last = tf.placeholder(tf.float32)
+                tf.summary.scalar('train_hitrate_last', self.train_hitrate_last)
+                self.train_mrr_last = tf.placeholder(tf.float32)
+                tf.summary.scalar('train_mrr_last', self.train_mrr_last)
```

Comment:
Ville vi egentlig ikke kunne lave en funktion der hedder noget create_summary_placeholders som tager en liste af placeholder værdier som liste (hitrate, mrr, etc) og så laver det her? bare for at slippe for vanvittigt mange linjer

bare en strøtanke btw

Reply:
Kan man godt det når man har med variabler at gøre? Tænker msåke det er unødvendigt besværligt for at spare på linjer

Reply:
Det kan du med eval() er jeg ret sikker på